test/e2e/operator_test.go

```diff
@@ -137,30 +146,49 @@ func TestDefaultContour(t *testing.T) {
+	if err := newDeployment(ctx, kclient, appName, cfg.SpecNs, testAppImage, testAppReplicas); err != nil {
+		t.Fatalf("failed to create deployment %s/%s: %v", cfg.SpecNs, appName, err)
 	}
-	t.Logf("created deployment %s/%s", specNs, appName)
+	t.Logf("created deployment %s/%s", cfg.SpecNs, appName)
 
-	if err := waitForDeploymentStatusConditions(ctx, kclient, 3*time.Minute, appName, specNs, expectedDeploymentConditions...); err != nil {
-		t.Fatalf("failed to observe expected status conditions for deployment %s/%s: %v", specNs, appName, err)
+	if err := waitForDeploymentStatusConditions(ctx, kclient, 3*time.Minute, appName, cfg.SpecNs, expectedDeploymentConditions...); err != nil {
+		t.Fatalf("failed to observe expected status conditions for deployment %s/%s: %v", cfg.SpecNs, appName, err)
 	}
-	t.Logf("observed expected status conditions for deployment %s/%s", specNs, appName)
+	t.Logf("observed expected status conditions for deployment %s/%s", cfg.SpecNs, appName)
 
-	if err := newClusterIPService(ctx, kclient, appName, specNs, 80, 8080); err != nil {
-		t.Fatalf("failed to create service %s/%s: %v", specNs, appName, err)
+	if err := newClusterIPService(ctx, kclient, appName, cfg.SpecNs, 80, 8080); err != nil {
+		t.Fatalf("failed to create service %s/%s: %v", cfg.SpecNs, appName, err)
 	}
-	t.Logf("created service %s/%s", specNs, appName)
+	t.Logf("created service %s/%s", cfg.SpecNs, appName)
 
-	if err := newIngress(ctx, kclient, appName, specNs, appName, 80); err != nil {
-		t.Fatalf("failed to create ingress %s/%s: %v", specNs, appName, err)
+	if err := newIngress(ctx, kclient, appName, cfg.SpecNs, appName, 80); err != nil {
+		t.Fatalf("failed to create ingress %s/%s: %v", cfg.SpecNs, appName, err)
 	}
-	t.Logf("created ingress %s/%s", specNs, appName)
+	t.Logf("created ingress %s/%s", cfg.SpecNs, appName)
 
-	if err := waitForHTTPResponse(testURL, 1*time.Minute); err != nil {
-		t.Fatalf("failed to receive http response for %q: %v", testURL, err)
+	// testURL is the url used to test e2e functionality.
+	testURL := "http://local.projectcontour.io/"
+
+	if isKind {
+		if err := waitForHTTPResponse(testURL, 1*time.Minute); err != nil {
+			t.Fatalf("failed to receive http response for %q: %v", testURL, err)
+		}
+		t.Logf("received http response for %q", testURL)
+	} else {
+		lb, err := waitForIngressLB(ctx, kclient, 1*time.Minute, cfg.SpecNs, appName)
+		if err != nil {
+			t.Fatalf("failed to get ingress %s/%s load-balancer: %v", cfg.SpecNs, appName, err)
+		}
+		t.Logf("ingress %s/%s has load-balancer %s", cfg.SpecNs, appName, lb)
+		testURL = fmt.Sprintf("http://%s/", lb)
+
+		// Curl the ingress from a client pod.
+		cliName := "test-client"
+		if err := podWaitForHTTPResponse(ctx, kclient, cfg.SpecNs, cliName, testURL, 3*time.Minute); err != nil {
+			t.Fatalf("failed to receive http response for %q: %v", testURL, err)
+		}
+		t.Logf("received http response for %q", testURL)
```

Comment:
it'd be nice to not have to repeat this whole block of code in each test, let's see if we can abstract it.

Reply:
@skriss I added this to https://github.com/projectcontour/contour-operator/issues/331 instead of refactoring since this PR is blocking https://github.com/projectcontour/contour-operator/pull/328 and we're so close to release.